Prescription Dialog and Issues List UI Improvement

robertdown opened the pull request:
The list of prescriptions and form for adding prescriptions have been updated to use bootstrap and more intuitive UI. I think the new form is clearer to understand as it heavily relies on placeholders. I've also swapped out regular select boxes with our Select2 option which can help speed up the data-entry process. The list screen has the table improved. 

The Issues page was also updated to put in properly styled tables and clean up the UI a little. Not a whole lot going on there, just some basic updates.

I did have to update Select2 from 4.0.3 to 4.0.8 but am not sure it's getting built properly - may need to ensure the updated file is actually making it. I'm tossing this into four.openemr.io/d in our up for grabs demo for input

bradymiller commented:
For issues gui; a bug i noted if that click on the code (this opens patient education popup), it also opens the edit issue popup.

bradymiller commented:
Up For Grabs demo for this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Eta_-_Up_For_Grabs_Demo

bradymiller commented on package.json:
```diff
@@ -74,7 +74,9 @@
     "numeral": "1.5.3",
-    "select2": "4.0.3",
+    "react": "16.3.2",
+    "react-dom": "16.3.2",
```
Are react and react-dom part if this PR?

robertdown replied:
Nope - I didn't bring that in

bradymiller replied:
Still unclear why react and react-dom are here (those are getting added in this PR; ie. git doesn't lie :) )

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -99,204 +86,240 @@
+            <label for="drug" class="control-label col-sm-2">{xlt t="Drug"}</label>
+            <div class="col-sm-7">
+                <div class="input-group">
+                    <input type="text" class="form-control" id="drug" name="drug" placeholder="{xlt t='e.x. Aspirin'}" value="{$prescription->drug|attr}">
```
sometimes using e.x. and sometimes using ex ?

robertdown replied:
Yeah, I'd prefer `e.x.` but the places I use `ex` are soley for spacing issues - if I used `e.x.` it would push the placeholder text out of the visible area of the textbox

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -99,204 +86,240 @@
+                <div class="input-group">
+                    <input type="text" class="form-control" id="drug" name="drug" placeholder="{xlt t='e.x. Aspirin'}" value="{$prescription->drug|attr}">
+                    <span class="input-group-btn">
+                        <button class="btn btn-default btn-search" tabindex="-1" type="button" onclick="$('#hiddendiv').show();document.getElementById('hidden_content').innerHTML='&lt;iframe src=&quot;controller.php?prescription&amp;lookup&amp;drug=&quot; width=&quot;100%&quot;height=&quot;75&quot; scrolling=&quot;no&quot; frameborder=&quot;no&quot;&gt;&lt;/iframe&gt;'">Search</button>
```
Translate Search

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -99,204 +86,240 @@
+        <div class="form-group {$dosageVisibility}">
+            <label for="size" class="control-label col-sm-2">{xlt t="Dosage"}</label>
+            <div class="col-sm-2">
+                <input type="text" class="form-control" name="size" id="size" placeholder="{xlt t='ex 81'}" maxlength="10" value="{$prescription->size|attr}">
```
use xla instead of xlt when within an html attribute. several other places to do this in this script.

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -99,204 +86,240 @@
+        <div class="form-group">
+            <div class="col-sm-offset-2 col-sm-10">
+                <a href="#" onclick="submitfun();" class="btn btn-default btn-sm btn-save">{xlt t="Save Prescription"}</a>
+                <button type="submit" name="disp_button"class='btn btn-sm btn-default btn-transmit'>{xla t='Save and Dispense'}</button>
```
Isn't above dependent on whether the dispensary module is turned on?

bradymiller commented on templates/prescription/general_list.html:
```diff
@@ -15,309 +16,286 @@
-        <p title="{xla t='Severity information will be missing if interaction is found'}"><a href="#">*{xlt t='Notice'}</a></p>
-        <div id="return_info">
-            {$INTERACTION}
+{else} {* No Prescriptions *}
```
translate

robertdown replied:
This is just an internal comment, no need to translate

bradymiller commented:
Just did a code review.

robertdown commented:
Ok @bradymiller I think I updated all the translation attribute fields, corrected the Search text to be translated, fixed the click issue with the code on the issues list and added in the proper condition for the "Save and Dispense" button to only display when appropriate. Ready for another review

bradymiller commented on interface/patient_file/summary/stats_full.php:
```diff
@@ -131,191 +123,203 @@ function newEncounter() {
                     }
-
-                    echo "  <td>" . text($row['referredby']) . "</td>\n";
-                    echo "  <td>" . text(oeFormatDateTime($row['modifydate'])) . "</td>\n";
```
Why need to remove the locale setting date formatting?

robertdown replied:
Whoops, since I submitted the PR that file developed merge conflicts, I resolved them but missed this. Pushing a fix now.

bradymiller commented on package.json:
```diff
@@ -74,7 +74,9 @@
+    "select2": "4.0.8",
     "select2-bootstrap-theme": "0.1.0-beta.10",
     "summernote": "0.8.2",
     "underscore": "1.8.3",
```
Placemarker for ongoing code review.

robertdown commented:
@bradymiller resolved date formatter

bradymiller commented:
@robertdown , Still the issue of react and react-dom being added in this PR. Planning to do extensive testing on this very soon.

robertdown commented:
I'll do another round of heavy testing, figure out where react came from and if it's testing well I'll probably merge this in soon
